packages/server/src/effects/logging.ts

```diff
@@ -0,0 +1,21 @@
+import type { SimulationState } from 'src/interfaces';
+import { Effect } from '../effect';
+import { assert } from 'assert-ts';
+import { spawn, Task } from 'effection';
+
+export const loggingEffect = (): Effect<SimulationState> => slice => function*(scope) {
+  let task: Task = yield spawn();
+
+  yield slice.slice('debug').forEach(function*(shouldLogErrors) {
+    if (shouldLogErrors) {
+      yield task.halt();
```

Comment:
I think we want to halt in all cases.

```suggestion
    yield task.halt();
    if (shouldLogErrors) {
```